### PR TITLE
test(e2e,playwright): ms-trophy-show.tsx

### PR DESCRIPTION
**Pull request by anand-harsh**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

 #51705 

image

### CODE REVIEW

File: e2e/ms-trophy-show.spec.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { test, expect } from '@playwright/test';
+import translations from '../client/i18n/locales/english/translations.json';
+
+const verifyTrophy = translations.buttons['verify-trophy'];
```
Comment (Ahmadkashif) [inline e2e/ms-trophy-show.spec.ts:4]:
```suggestion
const verifyTrophyButtonText = translations.buttons['verify-trophy'];
```

File: e2e/ms-trophy-show.spec.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+const askForHelpButton = translations.buttons['ask-for-help'];
```
Comment (Ahmadkashif) [inline e2e/ms-trophy-show.spec.ts:5]:
```suggestion
const askForHelpButtonText = translations.buttons['ask-for-help'];
```

File: e2e/ms-trophy-show.spec.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+});
+
+test('the page should render with correct title', async ({ page }) => {
+  await expect(page).toHaveTitle(
+    'Write Your First Code Using C# - Trophy - Write Your First Code Using C# | Learn | freeCodeCamp.org'
+  );
+});
+
+test('Correct Verify Trophy button', async ({ page }) => {
+  const askHelp = page.getByTestId('verify-trophy-button');
```
Comment (Ahmadkashif) [inline e2e/ms-trophy-show.spec.ts:24]:
```suggestion
  const askHelp = page.getByRole('button', { name: verifyTrophyButtonText })
    ).toBeVisible();
```

File: e2e/ms-trophy-show.spec.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  await expect(askHelp).toBeVisible();
```
Comment (Ahmadkashif) [inline e2e/ms-trophy-show.spec.ts:25]:
```suggestion
```

File: e2e/ms-trophy-show.spec.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  await expect(askHelp).toContainText(verifyTrophy);
```
Comment (Ahmadkashif) [inline e2e/ms-trophy-show.spec.ts:26]: if this fails, try .toHaveText() or the suitable call from documentation.

File: e2e/ms-trophy-show.spec.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+});
+
+test('Correct Ask for help button', async ({ page }) => {
+  const checkAnswer = page.getByTestId('ask-for-help-button');
```
Comment (Ahmadkashif) [inline e2e/ms-trophy-show.spec.ts:30]:
```suggestion
  const checkAnswer = page.getByRole('button', { name: askForHelpButtonText })
    ).toBeVisible();
```

File: e2e/ms-trophy-show.spec.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  await expect(checkAnswer).toBeVisible();
```
Comment (Ahmadkashif) [inline e2e/ms-trophy-show.spec.ts:31]:
```suggestion
```

**Review by Ahmadkashif — CHANGES_REQUESTED** [review]
Thank you for this amazing attempt @anand-harsh 💯 

I'll just add a few minor suggestions to try and improve this PR a little more.

**Comment by anand-harsh** [comment]
@Ahmadkashif I have tried resolving all issues. Kindly check and suggest if something more needed. Thanks for helping me with this.

**Review by Ahmadkashif — APPROVED** [review]
LGTM

**Review by naomi-lgbt — APPROVED** [review]

**Comment by naomi-lgbt** [comment]
Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 🎉

**Pull request merged**